src/webapp/src/auth/userSlice.ts

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { AuthStatuses, Login } from './types';
+import { signInUser, signOutUser } from './asyncAuthActions';
+
+const initialState: Login = {
+  user: null,
+  error: null,
+  authStatus: AuthStatuses.notLoggedIn,
+};
+export const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(signInUser.pending, (state) => {
+      state.authStatus = AuthStatuses.loading;
+    });
+    builder.addCase(signInUser.rejected, (state, action) => {
+      state.error = action.payload?.message || action.error?.message;
+      state.authStatus = AuthStatuses.notLoggedIn;
+    });
+    builder.addCase(signInUser.fulfilled, (state, { payload }) => {
+      state.user = payload;
```

Comment:
seeing this from laptop again, I think it will be better to have `uid`, `displayName` and `email` here instead of `user`